python/dgl/graphbolt/feature_store.py

```diff
@@ -0,0 +1,151 @@
+"""Feature store for GraphBolt."""
+import torch
+
+
+class FeatureStore:
+    r"""Base class for feature store."""
+
+    def __init__(self):
+        pass
+
+    def read_feature(self, key: str, ids: torch.Tensor = None):
+        """Read a feature from the feature store.
+
+        Parameters
+        ----------
+        key : str
+            The key that uniquely identifies the feature in the feature store.
+        ids : torch.Tensor, optional
+            The index of the feature. If specified, only the specified indices
+            of the feature are read. If None, the entire feature is returned.
+
+        Returns
+        -------
+        torch.Tensor
+            The read feature.
+        """
+        raise NotImplementedError
+
+    def update_feature(
+        self, key: str, value: torch.Tensor, ids: torch.Tensor = None
+    ):
+        """Update a feature in the feature store.
+
+        This function is used to update a feature in the feature store. The
+        feature is identified by a unique key, and its value is specified using
+        a tensor.
+
+        Parameters
+        ----------
+        key : str
+            The key that uniquely identifies the feature in the feature store.
+        value : torch.Tensor
+            The updated value of the feature.
+        ids : torch.Tensor, optional
+            The index of the feature. If specified, only the specified indices
+            of the feature will be updated. If None, the entire feature will be
```

Comment:
This comment is confusing. My understand is, the value will be n vectors, and ids will be n id, and row i of value will be update to featurestore[key][id[i]].

Reply:
Which part is confusing? If `ids` is given, it works exactly as what you described.

Reply:
I changed the description.

```
        ids : torch.Tensor, optional
            The indices of the feature to update. If specified, only the
            specified indices of the feature will be updated. For the feature,
            the `ids[i]` row is updated to `value[i]`. So the indices and value
            must have the same length. If None, the entire feature will be
            updated.
```